### PR TITLE
Update developer documentation

### PR DESCRIPTION
**Pull request by glass-ships**
#### Summary of work

This PR aims to update the developer documentation to:
- include additional useful information
- remove/replace outdated information

#### Purpose of work

*There is no associated issue.*

#### Further detail of work

In particular, information regarding dependencies is lacking at current (at least for Ubuntu systems, especially those which may have been newly/recently provisioned).  
Namely:
- GCC / `build-essential`
- OpenGL

Also added `.venv` to git ignore, as this directory is created by default when running `conda env create ...` within mantid.

### Additional Notes/Tasks

- It is worth ensuring that the OSX and Windows with Conda, as well as the no-conda pages include equivalent information, as I can only speak to my experience in a Ubuntu environment. 
- Somewhat unrelated, but I would also recommend including helpful information about setting up and using the restructuredtest / esbonio VSCode plugins; functionality seems to have changed recently, making live previews pretty finnicky) 

There may be other updates/changes that are worth making, but these were the main things I noticed going through my first build of Mantid which other new developers may find useful. 

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: dev-docs/source/GettingStarted/GettingStarted.rst [hunk 1]
```diff
@@ -46,7 +53,9 @@ If you need further help, ask another developer at the facility how to configure
 
 Building Mantid
 ###############
-See :ref:`BuildingWithCMake` for information about building Mantid.
+
+| From this point, you can proceed to build Mantid. See :ref:`BuildingWithCMake` for information.
+| For developers at ISIS, continue reading for information on accessing the archive.
```
Comment (robertapplin) [inline dev-docs/source/GettingStarted/GettingStarted.rst:58]: Has this been indented by accident?

File: dev-docs/source/GettingStarted/GettingStarted.rst [hunk 2]
```diff
@@ -15,17 +15,24 @@ Getting Started
...
-The Choice
-##########
+Dependencies
+############
 
-With the addition of Conda to our workflow we at present have two different methods of Getting Started with mantid
-development. Therefore you can choose which method to setup with (We recommend Conda as it will be the only supported
+The dependencies required to build Mantid will vary depending on your platform and the method you choose to setup with. At the very least you will need:
+
+* `Git <https://git-scm.com/>`_
```
Comment (robertapplin) [inline dev-docs/source/GettingStarted/GettingStarted.rst:23]: This is more of a suggestion, so feel free to push back! I'm not sure if its helpful to list git as a dependency here, as its already listed in the "Getting started" instructions for each of the OS's

**Review by robertapplin — CHANGES_REQUESTED** [review]
These are some nice additionss/changes to the getting started documentation pages. I think @thomashampson will add a few suggestions on this PR too

File: dev-docs/source/GettingStarted/GettingStartedCondaLinux.rst [hunk 3]
```diff
@@ -4,10 +4,19 @@
...
+Install system dependencies
+---------------------------
+Mantid requires a number of system dependencies to be installed via your system's package manager before building. These include:
+
+* GCC, g++, make, etc.
+    * For Ubuntu/Debian based systems install using apt with the following command: ``sudo apt install build-essential``
+    * For CentOS/RHEL based systems install using yum with the following command: ``sudo yum groupinstall 'Development Tools'``
+* OpenGL
+    * For Ubuntu/Debian based systems install using apt with the following command: ``sudo apt install libgl1-mesa-dev``
+    * For CentOS/RHEL based systems install using yum with the following command: ``sudo yum install mesa-libGL-devel``
```
Comment (thomashampson) [inline dev-docs/source/GettingStarted/GettingStartedCondaLinux.rst:16]: These dependencies are installed in the conda environment, so all you really need is git.

**Review by thomashampson — CHANGES_REQUESTED** [review]
You shouldn't need to install anything manually except for git when setting up your linux development environment.

File: dev-docs/source/GettingStarted/GettingStarted.rst (hunk 1 above)
Comment (glass-ships) [reply dev-docs/source/GettingStarted/GettingStarted.rst:58]: It shouldn't be indented, I *think* this just forces a new line without adding paragraph spacing. Totally optional, just helps readability imo

File: dev-docs/source/GettingStarted/GettingStarted.rst [hunk 4]
```diff
@@ -15,17 +15,24 @@ Getting Started
...
+* `CMake <https://cmake.org/download/>`_ >= 3.15
+* OpenGL
+* GCC
```
[the remark this answers is not in the document]
Comment (glass-ships) [reply dev-docs/source/GettingStarted/GettingStarted.rst:26]: Yeah this is something I've been trying to sort out the last couple days. @peterfpeterson mentioned this was the case, but it seems like something is wrong with my environment. If I try building without installing GCC and OpenGL at the system level, I get: 
```                                                                   
Preset CMake variables:

  CMAKE_CXX_FLAGS_DEBUG="-g -Og"
  CMAKE_PREFIX_PATH="/home/ge2/mambaforge/envs/mantid-developer"

-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:5 (project):
  The CMAKE_CXX_COMPILER:

    /usr/bin/c++

  is not a full path to an existing compiler tool.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
```

this is using the command `cmake ./mantid --preset=linux DCMAKE_BUILD_TYPE=RelWithDebInfo -B ./builds/test`

**Review by robertapplin — APPROVED** [review]

**Review by thomashampson — APPROVED** [review]

**Pull request merged**